Calculate version using setuptools scm (#534)
canonical/checkbox: checkbox-core-snap/README.md

# Building checkbox-core-snap
This snap is the core reusable part of checkbox, it includes all the utilities
and some core providers.

## Building guide

> This section convers how to build the snap in a normal settings. 

Reviewer: ```suggestion
This section covers how to build the snap
```

Reviewer: (correct would be `setting`, not `settings`, but it is also redundant)

via either the `multipass` or the `lxd` backend. This guide will focus on the latter
but with small adjustments can be used for the former. Also, this guide will
build the `series22` snap, adjust the commands to build any other.

In order to build the snap you are going to need the following:
1. **snapcraft:** Installed via `snap install snapcraft --classic` or 
`snap install snapcraft --classic --channel=4.x` for series16
2. **python3:** Installed via `apt` or any other packaging solution
3. **setuptools_scm:** Installed via pip
4. **rsync:** Installed via apt or any other packaging solution

> After installing all the dependencies do the following:

Reviewer: Perhaps explain what `prepare.sh` is about. For example:

"After installing all the dependencies and cloning the repo, to output the snapcraft.yaml for `core22` as an example, do the following:"

```
> git clone https://github.com/canonical/checkbox
> cd checkbox/checkbox-core-snap/
> ./prepare.sh series22
> cd series22
```

Finally build the snap with:
```
> snapcraft --use-lxd
```

If the build has failed, check either the **Building guide for debugging**, if
it completed succesfully, refer to **Testing the build**.

## Testing the build

> To test the build one must install it and see if the content is correct. This

Reviewer: ```suggestion
To test the build one must install it and see if the content is correct.
```

> process is not perfect, but one way to do it is the following:

Reviewer: ```suggestion
One way to do it is the following:
```

Reviewer: (cruft)

```bash
> snap install checkbox22_(version)_(arch).snap --dangerous
```

> Now we have installed the core snap, but we can not use it directly. To use it

Reviewer: ```suggestion
Now we have installed the core snap, but we cannot use it directly. To use it
```

> we need a simple user snap. 

Author: I can change this for front-end snap if it is clearer, what I want to say it that we need a snap that uses the core

> we need a simple user snap. 

Reviewer: Frontend makes sense indeed. If there's a place for this in the user docs, do add that too.

> we need a simple user snap. 

Author: Updated here, given that this whole readme should be ported to the docs as well I think that the operation of documenting the whole frontend / core business should be done there. Also, we should consider renaming core to something like runtime, but that is unrelated to this PR imo

```bash
# Note: install the correct channel for your series
> snap install checkbox --classic --channel=22.04
```
Now you can test the checkbox snap, you can try to run test plans as follows.
For example, you may want to run the smoke test plan.
```
> checkbox.checkbox_cli
```

> ## Builing guide for debugging

Reviewer: (Typo + the term "guide" weirds me out a bit since I might call the whole set of of documentation a "guide")
